helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java

```diff
@@ -624,19 +624,28 @@ public void rollingUpdate(final DeploymentGroup deploymentGroup,
       throws DeploymentGroupDoesNotExistException, JobDoesNotExistException {
     checkNotNull(deploymentGroup, "deploymentGroup");
 
-    log.info("preparing to initiate rolling-update on deployment-group: name={}, jobId={}",
-        deploymentGroup.getName(), jobId);
+    final Job job = getJob(jobId);
+    if (job == null) {
+      throw new JobDoesNotExistException(jobId);
+    }
+
+    final RolloutOptions mergedRolloutOptions =
```

Comment:
As I mentioned before, the downside I see so far with this approach is that jobs can be created with null values for rollout options. They'll exist on disk with null values for these options. When you inspect them, they'll output null values. Then when you start a rolling-update, the master will fill in the defaults. This is different behavior from the other options in the job config. The state they are in when created and on disk is the final configuration at deployment time.

Reply:
On the other hand, if we fill in the null values on job create, the config file used to create the job will differ from the config stored in zk. Though I suppose that's strictly true even when we keep the null values, since our JSON reader allows missing fields while the writer writes all fields.

Field in config / cli|Behavior|Field in Zookeeper / `helios inspect`
-|-|-
Present|Fill with `null`|As given in config / cli
Present|Fill with default|As given in config / cli
Absent|Fill with `null`|Strictly differs but logically equivalent
Absent|Fill with default|Differs from config and may be unexpected

Neither alternative to the handing of absent fields is clearly preferable in my view. I'd say that filling with defaults gives the most information, and can be clear enough if we document the defaults and their application during job creation.

Reply:
> I'd say that filling with defaults gives the most information, and can be clear enough if we document the defaults and their application during job creation.

agreed